runner/clireporter.ts

```diff
@@ -0,0 +1,304 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+import * as _ from 'lodash';
+import * as chalk from 'chalk';
+import * as cleankill from 'cleankill';
+import * as stacky from 'stacky';
+import * as util from 'util';
+import * as events from 'events';
+import * as config from './config';
+import {BrowserDef, Stats} from './browserrunner';
+import * as context from './context';
+import * as tty from 'tty';
+
+const STACKY_CONFIG = {
+  indent: '    ',
+  locationStrip: [
+    /^https?:\/\/[^\/]+/,
+    /\?[\d\.]+$/,
+  ],
+  unimportantLocation: [
+    /^\/web-component-tester\//,
+  ]
+};
+
+type State = 'passing' | 'pending' | 'failing' | 'unknown' | 'error';
+type Formatter = (value: string) => string;
+
+const STATE_ICONS = {
+  passing: '✓',
+  pending: '✖',
+  failing: '✖',
+  unknown: '?',
+};
+
+const STATE_COLORS: {[state: string]: Formatter} = {
+  passing: chalk.green,
+  pending: chalk.yellow,
+  failing: chalk.red,
+  unknown: chalk.red,
+  error:   chalk.red,
+};
+
+
+const SHORT = {
+  'internet explorer': 'IE',
+};
+
+const BROWSER_PAD = 24;
+const STATUS_PAD  = 38;
+
+interface TestEndData {
+  state: 'passing'|'failing'|'pending'|'unknown';
+  test: string[];
```

Comment:
if you can add any comments... I'm wondering why `test` is a `string[]`, what's in it?